pkg/runtime/inject/inject_test.go

```diff
@@ -0,0 +1,193 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package inject
+
+import (
+	"fmt"
+
+	"github.com/kubernetes-sigs/controller-runtime/pkg/cache"
+	"github.com/kubernetes-sigs/controller-runtime/pkg/cache/informertest"
+	"github.com/kubernetes-sigs/controller-runtime/pkg/client"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+)
+
+var instance testSource
+var neg struct{}
+var _ = Describe("runtime inject", func() {
+
+	It("should set informers", func() {
+		instance := testSource{
+			cache: func(c cache.Cache) error {
+				if c != nil {
```

Comment:
We should validate that the injected argument is the same as the one you provided.